gfx/common/vulkan_common.c

```diff
@@ -1896,50 +2030,69 @@ bool vulkan_context_init(gfx_ctx_vulkan_data_t *vk,
       return false;
    }
 
-   use_instance_ext                  = vulkan_find_instance_extensions(instance_extensions, ext_count);
-
-   app.pApplicationName              = msg_hash_to_str(MSG_PROGRAM);
-   app.applicationVersion            = 0;
-   app.pEngineName                   = msg_hash_to_str(MSG_PROGRAM);
-   app.engineVersion                 = 0;
-   app.apiVersion                    = VK_MAKE_VERSION(1, 0, 18);
-
-   info.pApplicationInfo             = &app;
-   info.enabledExtensionCount        = use_instance_ext ? ext_count : 0;
-   info.ppEnabledExtensionNames      = use_instance_ext ? instance_extensions : NULL;
-#ifdef VULKAN_DEBUG
-   info.enabledLayerCount            = ARRAY_SIZE(instance_layers);
-   info.ppEnabledLayerNames          = instance_layers;
-#endif
+   app.pApplicationName   = msg_hash_to_str(MSG_PROGRAM);
+   app.applicationVersion = 0;
+   app.pEngineName        = msg_hash_to_str(MSG_PROGRAM);
+   app.engineVersion      = 0;
+   app.apiVersion         = VK_API_VERSION_1_0;
 
    if (iface && iface->get_application_info)
    {
-      info.pApplicationInfo = iface->get_application_info();
+      app = *iface->get_application_info();
```

Comment:
I'd resort to the default `VkInstance` creation behavior if `get_application_info` returns `NULL`, as I could see a core leaving instance creation details to the frontend in some circumstances.

Reply:
Cleaned it up a bit now. Made it mandatory in v2 and left v1 loose, since some cores might have relied on this behavior.